Libraries/LibGUI/Action.h

```diff
@@ -132,6 +132,9 @@ class Action final : public Core::Object {
     }
     void set_checked(bool);
 
+    bool swallow_key_event_when_disabled() const { return m_swallow_key_event_when_disabled; }
+    void set_swallow_key_event_when_disabled(bool swallow) { m_swallow_key_event_when_disabled = swallow; }
```

Comment:
This is a very niche situation but I think you've come up with a decent way of handling it.